src/Log4Net/NewRelic.LogEnrichers.Log4Net/NewRelicLayout.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using log4net.Core;
+using log4net.Layout;
+using Newtonsoft.Json;
+
+namespace NewRelic.LogEnrichers.Log4Net
+{
+    public class NewRelicLayout : LayoutSkeleton
+    {
+        private const string LinkingMetadataKey = "newrelic.linkingmetadata";
+
+        public NewRelicLayout()
+        {
+            base.IgnoresException = false;
+        }
+        public override void ActivateOptions()
+        {
+        }
+
+        public override void Format(TextWriter writer, LoggingEvent loggingEvent)
+        {
+            var dic = new Dictionary<string, object>();
+
+            SetInstrinsics(dic, loggingEvent);
+            SetExceptionData(dic, loggingEvent);
+            SetUserProperties(dic, loggingEvent);
+            SetLinkMetaData(dic, loggingEvent);
+
+            writer.WriteLine(JsonConvert.SerializeObject(dic));
+        }
+
+        void SetInstrinsics(Dictionary<string, object> dictionary, LoggingEvent loggingEvent)
```

Comment:
This is just a coding style comment, but I think we should be more explicit about the visibility of these methods so that we are consistent with the other code in the this repository.